specs/datapackages/README.md

```diff
@@ -0,0 +1,130 @@
+
+Perhaps to make it more accessible we can link to this page from the dataset creation form (from the top bar or the left hand column).
+
+### Potential issues
+
+Identifiers: Data Packages are solely identified by the `name` field, which according to the Data Package spec:
+
+> [is a] short url-usable (and preferably human-readable) name of the package. This MUST be lower-case and contain only alphanumeric characters along with ., _ or - characters. It will function as a unique identifier and therefore SHOULD be unique in relation to any registry in which this package will be deposited (and preferably globally unique).
+
+We can not assume that these identifiers will be unique and that they will not clash with existing CKAN datasets. We can append stuff to the name but then we need to consider what happens when we reupload the same Data Package (ie to update the dataset).
+
+**Q:** Do we upload the files described on the Data Package to the CKAN server?
+I would say not by default. As long as the files are publicly accessible the DataPusher can get them into the DataStore.
+
+**Q:** Do we keep a copy of the original `datapackage.json` ?
+I think yes. Trying to store all fields as extras and replicate them back when generating the datapackage back would be too much hassle. Also we would need to keep adding fields as the spec changes and handle arbitrary fields added by publishers. An easier approach might be outputting back what was uploaded, but with the fields that are actually mapped to CKAN (title, description, keywords etc) updated, as these might have been modified in CKAN.
+
+## Exporting Data Packages
+
+### Implementation
+
+Accessing `/dataset/{id}/datapackage.json` will output the generated Data Package for this dataset. Similarly as above, we'll have a `datapackage_export` action (with the same auth as `package_show`) and a controller to handle requests.
+
+### Potential issues
+
+See point above regarding outputting back custom fields.
+
+I think that the JSON files could be generated on the fly, but if performance is an issue the Data Package JSON blobs for each dataset could be generated before hand (using `after_create` and `after_update`) and stored in the DB or even as a field in the Solr index.
+
+**Q:** So far we've only considered giving back the `datapackage.json` file, but would it make sense to have an extra endpoint that returns a zip of the `datapackage.json` plus the data files?
+
+## Resource schemas
+
+
+### Implementation
+
+Add new core fields to the resource model:
+    * `schema`: a JSON object describing the structure of the resource file, or a URL linking to the schema.
+    * `schemaType`: a string or URI that specifies what is the specification for the schema. 
+    * `schemaVersion`: the version of the spec being used
```

Comment:
The names are fine, although I'm not sure about the camelcase - maybe `schema_type` and `schema_version` to match things all the others in CKAN like `owner_org`?